sycl/source/handler.cpp

```diff
@@ -59,20 +84,57 @@ getOrInsertHandlerKernelBundleCommon(
 
     detail::ExtendedMemberT EMember = {
         detail::ExtendedMembersType::HANDLER_KERNEL_BUNDLE, KernelBundleImpPtr};
-
-    ExendedMembersVec->push_back(EMember);
+    ExtendedMembersVec->push_back(EMember);
   }
 
   return KernelBundleImpPtr;
 }
 
-// Returns a shared_ptr to kernel_bundle stored in the extended members vector.
-// If there is no kernel_bundle created:
-// returns newly created kernel_bundle if Insert is true
-// returns shared_ptr(nullptr) if Insert is false
-std::shared_ptr<detail::kernel_bundle_impl>
-handler::getOrInsertHandlerKernelBundle(bool Insert) const {
+// If the submission state is SPEC_CONST_SET_STATE this function returns false.
+// Otherwise it sets the submission state to EXPLICIT_KERNEL_BUNDLE_STATE and
+// returns true.
+bool handler::setStateExplicitKernel() {
+  std::lock_guard<std::mutex> Lock(
+      detail::GlobalHandler::instance().getHandlerExtendedMembersMutex());
+
+  assert(!MSharedPtrStorage.empty());
 
+  std::shared_ptr<std::vector<detail::ExtendedMemberT>> ExendedMembersVec =
+      detail::convertToExtendedMembers(MSharedPtrStorage[0]);
+
+  auto HandlerImpl = getHandlerImpl(ExendedMembersVec);
```

Comment:
Suggest reorganizing methods that access `handle_impl` to make them more similar to our "pimpl" like classes:
```
bool handler::setStateExplicitKernel() {
  getHandlerImpl()->setStateExplicitKernel();
}
```

Reply:
Agreed. This has been done.